### PR TITLE
Fix swiper-isearch breaks ivy-resume

### PR DESCRIPTION
**Pull request by amosbird**
https://github.com/abo-abo/swiper/issues/2083

### CODE REVIEW

File: ivy.el [hunk 1]
```diff
@@ -1082,7 +1082,10 @@ If the text hasn't changed as a result, forward to `ivy-alt-done'."
     (when (eq (ivy-state-caller ivy-last) 'swiper)
       (switch-to-buffer (ivy-state-buffer ivy-last)))
     (with-current-buffer (ivy-state-buffer ivy-last)
-      (let ((default-directory (ivy-state-directory ivy-last)))
+      (let ((default-directory (ivy-state-directory ivy-last))
+            (ivy-format-function (if (eq (ivy-state-caller ivy-last) 'swiper-isearch)
+                                     #'swiper-isearch-format-function
+                                   #'ivy-format-function)))
```
Comment (basil-conto) [inline ivy.el]: Is there no other way to avoid hard-coding this?

**Comment by amosbird** [comment]
There is also a case for `swiper' too.

**Comment by basil-conto** [comment]
> There is also a case for `swiper' too.

WDYM? Why shouldn't we look for better alternatives to adding more hard-coded collection checks?

**Comment by abo-abo** [comment]
Thanks, all. I went with a generic solution that introduces `ivy-format-functions-alist` and makes `ivy-format-function`.

Please review and test if it fixes the issue.

**Comment by manuel-uberti** [comment]
I can resume a `swiper-isearch` now. Thanks.

**Pull request closed without merging**